### ISSUE TITLE
Eclipse 4.11 compatible plug-in version

### ISSUE THREAD

**Issue by dfht** [gwt-plugins/gwt-eclipse-plugin#393]
There doesn't seem to be one. Is one planned?
Probably only needs a few fixes due to stuff moved in JDT etc., but not sure...

**Comment by CrushaKRool** [comment]
I updated the code base in my fork to support Eclipse 4.12 (which indeed meant fixing up some compile errors due to changes in JDT) and then published a local repo from which I could install my version into Eclipse.

However, I am actually getting reports from some of my coworkers that they constantly run into issues when they run this new GWT plugin in SuperDevMode and make code changes with the SDBG source mapping debugger attached. The JDK process that is responsible for the SuperDevMode just keeps eating more and more memory (we already assigned 4GB to the JDK, but that still doesn't seem enough) until it eventually gets to a point where it just gets stuck and throws random JavaScript errors about undefined methods and unknown stacktraces in the browser.
Reportedly this behavior did not happen in an older Eclipse 4.9 with an old version of the plugin that was actually built against the JDT sources for Eclipse Mars, I believe.

Long story short: I am not sure whether I actually broke something in my update or whether there is some change in the JDT that prevents the GWT compiler from being correctly notified about changed sources during debug time. Or perhaps it's just something completely different entirely. Might even have been an issue in 4.9 as well and we just never noticed.

**Comment by CrushaKRool** [comment]
I was trying to wrap my changes into a pull request to update the libraries and code base to Eclipse 4.15 (which is not the most recent due to their quarterly release cycle, but it's what we are currently using in production).

Unfortunately that's when I noticed that the plugin no longer passes the deployment process when I do that. I did not notice the issue before because I accidentally still told it to deploy with the Eclipse Oxygen target, even though the libraries referenced by the code were for Eclipse 2020-03.

The problem I am running into is that some referenced dependencies can no longer be resolved from their newer repositories:
```
[ERROR] Cannot resolve project dependencies:
[ERROR]   You requested to install 'org.eclipse.jst.web_sdk.feature.feature.group 0.0.0' but it could not be found
[ERROR] 
[ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[ERROR] Cannot resolve dependencies of MavenProject: com.gwtplugins.eclipse:com.gwtplugins.gwt.eclipse.wtp.test:3.0.0-SNAPSHOT @ E:\WorkingCopies\gwt-eclipse-plugin\plugins\com.gwtplugins.gwt.eclipse.wtp.test\pom.xml: See log for details -> [Help 1]
```

That happens if I use version R3.17.0 of the Eclipse Web Tools Platform.
If I use the old R3.9.0 version, it does not seem to run into the same issue but instead reports compatibility issues because some of the other transitive dependencies are too new to be used with that old version.
Perhaps the dependencies have been deprecated or removed. Or it could be some configuration issue on my part.
When I exclude that dependency, I just get the same issue on the next one from that repo.

So far, this issue only seems to affect the unit testing part of `com.gwtplugins.gwt.eclipse.wtp.test`, so I am not sure if it actually matters for production use. Still, it leaves me wondering whether there is any merit to providing a pull request if I know that it can't be deployed in its current form.
(@branflake2267 , would an incomplete pull request be of use to you in any way? I'd like to help getting the official plugin back up to speed for current Eclipse versions, but I am not quite sure I can properly resolve this issue due to my unfamiliarity with the plugin ecosystem.)

**Comment by keinhaar** [comment]
eclipse 4.11 is very outdated.
Plugin runs with current versions of eclipse > 2023-09.
